example/aws/composition.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: shell-example
+spec:
+  compositeTypeRef:
+    apiVersion: example.crossplane.io/v1
+    kind: XR
+  mode: Pipeline
+  pipeline:
+    - step: pull-extra-resources
+      functionRef:
+        name: function-extra-resources
+      input:
+        apiVersion: extra-resources.fn.crossplane.io/v1beta1
+        kind: Input
+        spec:
+          extraResources:
+            - kind: ProviderConfig
+              into: ProviderConfig
+              apiVersion: aws.upbound.io/v1beta1
+              type: Selector
+              selector:
+                maxMatch: 1
+                minMatch: 1
+                matchLabels:
+                - key: account
+                  type: Value
+                  value: demo
+    - step: shell
+      functionRef:
+        name: function-shell
+      input:
+        apiVersion: shell.fn.crossplane.io/v1alpha1
+        kind: Parameters
+        shellEnvVars:
+          - key: AWS_ROLE_ARN
+            value: arn:aws:iam::00000001:role/Crossplane
+          - key: AWS_ASSUME_ROLE_ARN
+            valueRef: "context[apiextensions.crossplane.io/extra-resources].ProviderConfig[0].spec.assumeRoleChain[0].roleARN"
```

Comment:
It won't change because I set the following https://github.com/clementblaise/function-shell/blob/158f342c28165c89c0bda813240b1f59254d0670/example/aws/composition.yaml#L24-L25.
In case no ProviderConfig is found, the composition will error out.

Reply:
Super. Thanks.